packages/table/src/table.tsx

```diff
@@ -491,6 +491,74 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
     // ================
 
     /**
+     * __Experimental!__ Resizes all rows in the table to the approximate
+     * maximum height of wrapped cell content in each row. Works best when each
+     * cell contains plain text of a consistent font style (though font style
+     * may vary between cells). Since this function uses approximate
+     * measurements, results may not be perfect.
+     *
+     * Approximation parameters can be configured for the entire table or on a
+     * per-cell basis. Default values are fine-tuned to work well with default
+     * Table font styles.
+     *
+     * @param getCellText the text content of the cell
+     * @param getApproximateCharWidth the approximate width (in pixels) of an
+     * average character of text
+     * @param getApproximateLineHeight the approximate height (in pixels) of an
+     * average line of text
+     * @param getCellHorizontalPadding the sum of horizontal paddings (in
+     * pixels) from the left __and__ right sides of the cell
+     * @param getNumBufferLines the number of extra lines to add in case the
+     * calculation is imperfect
+     */
+    public resizeRowsByApproximateHeight(
+        getCellText: ICellMapper<string>,
+        getApproximateCharWidth: number | ICellMapper<number> = 8,
+        getApproximateLineHeight: number | ICellMapper<number> = 18,
+        getCellHorizontalPadding: number | ICellMapper<number> = 2 * Locator.CELL_HORIZONTAL_PADDING,
+        getNumBufferLines: number | ICellMapper<number> = 1,
```

Comment:
well dang that's a lot of args! how would you feel about accepting a single object with a bunch of required keys? could drop the `get` prefix with the object, since you'd have to write out the keys each time.

I'd also accept the argument that this is fine in typescript and we should just forge ahead.

Reply:
that would allow you to specify a specific override without having to place in the right order in args. still, config object args are somewhat awkward.

awkward because you have to check if the obj is null, then check null for each thing. lodash/defaults would help here.

Reply:
Single object sounds fine, but I don't mind too much either way. What would the object look like? Same as the above, except with the param names as keys?